cogs/birthday/birthday.py

```diff
@@ -519,46 +550,111 @@ async def setSelfBirthday(self, ctx: Context, *, birthday: MonthDayConverter):
                 f"Type {bold('`yes`', escape_formatting=False)} to confirm."
             )
 
-            await ctx.author.send(f"{headerWarn}: {confirmationStr}")
+            try:
 
-            def check(msg: discord.Message):
-                return msg.author == ctx.author and msg.channel == ctx.author.dm_channel
+                def check(msg: discord.Message):
+                    return msg.author == ctx.author and msg.channel == ctx.author.dm_channel
+
+                await ctx.author.send(f"{headerWarn}: {confirmationStr}")
+
+                try:
+                    response = await self.bot.wait_for("message", timeout=30.0, check=check)
+                except asyncio.TimeoutError:
+                    await ctx.author.send(
+                        f"{headerBad}: You took too long. Not setting your birthday."
+                    )
+                    return
+
+                if response.content.lower() != "yes":
+                    await ctx.author.send(f"{headerBad}: Declined. Not setting your birthday.")
+                    return
+
+                await birthdayConfig.get_attr(KEY_BDAY_MONTH).set(birthday.month)
+                await birthdayConfig.get_attr(KEY_BDAY_DAY).set(birthday.day)
+                await birthdayConfig.get_attr(KEY_ADDED_BEFORE).set(True)
 
-            try:
-                response = await self.bot.wait_for("message", timeout=30.0, check=check)
-            except asyncio.TimeoutError:
-                # hide the birthday portion within the previous message
                 await ctx.author.send(
-                    f"{headerBad}: You took too long. Not setting your birthday."
+                    f"{headerGood}: Successfully set your birthday to "
+                    f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}."
+                )
+
+                self.logger.info(
+                    "%s#%s (%s) added their birthday as %s",
+                    ctx.author.name,
+                    ctx.author.discriminator,
+                    ctx.author.id,
+                    birthdayStr,
                 )
+
+                # explicitly check to see if user should be added to role, if the month
+                # and day just so happen to be the same as it is now.
+                await self.checkBirthday()
                 return
+            except discord.Forbidden:
+
+                def check(msg: discord.Message):
+                    return msg.author == ctx.author and msg.channel == ctx.channel
```

Comment:
+1 same sentiment as tatsu on putting this close to the try block.

Reply:
All checks have been moved right before their usages